### ISSUE TITLE
filter linebreaks(br) and escaping

### ISSUE THREAD

**Issue by pierrre** [flosch/pongo2#222]
When you use the `linebreaks(br)` filter, the input is usually pure text, not HTML.
If we replace `\n` with `<br \>`, then the output becomes HTML.
So it must be flagged as safe in the template.
Therefore, we need to do `value | linebreaksbr | safe` in order to handle the inserted br correctly, which is not very user-friendly.
But now, the input is considered as safe, which is not necessarily true, because it could contain special HTML character.
So the input must also be escaped manually: `value | escape | linebreaksbr | safe`: even less user-friendly.
Wouldn't it be simpler to call `html.EscapeString()` at the beginning of the filter, and `AsSafeValue()` at the end ?

**Comment by pierrre** [comment]
Yes it's not backward compatible, but I think it makes more sense.

**Comment by router0mail** [comment]
Confirmed this is still live at current HEAD while auditing pongo2, and found the changelog's claimed fix didn't actually address it — sharing the details in case it helps close this out.

CHANGELOG.md (commit b1851e5) says linebreaks/linebreaksbr were "marked as safe to prevent double-escaping with autoescape" — but reading that commit, it only added the `AsSafeValue()` call to the filter's output. It never added an actual HTML-escaping step on the raw input first. So both filters still emit attacker-controlled input completely unescaped:

```go
tpl, _ := pongo2.FromString(`{{ data|linebreaksbr }}`)
out, _ := tpl.Execute(pongo2.Context{"data": "line1<script>alert(1)</script>\nline2"})
// out contains the raw <script> tag, unescaped
```

Same for `linebreaks`. Root cause is `filterLinebreaks`/`filterLinebreaksbr` in `filters_builtin.go` (lines 1232/1244 and 1272/1274) calling `AsSafeValue()` on the output without ever routing the input through the engine's own escaping step first.

While digging into this I also found two more built-in filters with the exact same bug pattern that aren't mentioned in this issue: `truncatechars_html` and `truncatewords_html` (`filters_builtin.go` lines 408/419/442 and 492/540). Same root cause, same fix needed — escape the raw input before calling `AsSafeValue()`, matching how `urlize`/`urlizetrunc` already do it correctly.

Happy to share a full PoC test file if useful for a fix PR.
